testing/trino-testing/src/main/java/io/trino/testing/BaseConnectorTest.java

```diff
@@ -601,6 +606,95 @@ public void testExplainAnalyze()
         assertExplainAnalyze("EXPLAIN ANALYZE SHOW SESSION");
     }
 
+    @Test
+    public void testMaterializedViewsMetadata()
```

Comment:
Test should create multiple MVs and verify output for queries with filters on
1. catalog_name
2. catalog_name and schema_name
3. catalog_name, schema_name and name
4. schema_name LIKE '%" + schemaName + "%'"
5. name LIKE '%" + viewName + "%'

Reply:
@raunaqmorarka I also think that we should verify that the filtering is actually applied in `MaterializedViewSystemTable`, since the engine could be filtering. I've verified that `MaterializedViewSystemTable` actually applies the filtering manually, but since we don't use any mock/stub framework, do you know a good way to verify this?